### DISCUSSION TITLE
Bumping Fees in RTL with Core Lightning

### DISCUSSION THREAD

**Discussion by jairunet** [Ride-The-Lightning/RTL#1189] [C-Lightning]
Hello RTL community,

I am trying to figure out how to correctly bump the fees on a lightning channel opened using `RTL` + `Core Lightning`.  For example, when I go to the pending lightning channels section, I get this message:

![SCR-20230210-anj](https://user-images.githubusercontent.com/5609312/217956298-47637a51-58dd-4dda-9a15-b26872f0beed.png)

I was looking into the `blockstream block explorer` and also the `mempool.space` site and I was not able to easily find the information requested in `RTL` to bump the fee:

![SCR-20230210-any](https://user-images.githubusercontent.com/5609312/217957068-3d11c89b-3c9e-48cb-b935-c2278bab5ab8.png)

If there is documentation with references about this will be great to know.  I appreciate any help on this.

Sincerely,

**Comment by jairunet** [comment]
Hello RTL Community,

I was able to figure out where to find the index value of the pending channel, then use it together with the new fee and apply it to the `ancestor` transaction ID, then noticed that a new CPFP transaction ID gets created attached to the previous one but with the CPFP label and a new fee as well.

![image](https://user-images.githubusercontent.com/5609312/218132605-58871c9e-dabe-4bf1-962f-abab518119c1.png)


I will be creating a tutorial for it as I am sure I am not the only one that would have this need from time to time.

Thank you for all the great software and proof of work.

Sincerely,

**Reply by saubyk** [reply]
Hi @jairunet 
We tried to make it as clear as possible in the application documentation, but this feature is still not well suited for users who don't understand how Bitcoin transactions work.
The transaction which can be used to bump up the pending transaction, is the *change output*. The key is in finding the index value of that change output

**Reply by saubyk** [reply]
Also, if you can create a tutorial, to explain the feature better, that would definitely be helpful :-)

**Reply by jairunet** [reply]
Hello @saubyk,

And thank you for getting back to me on this, I will find the time to create a tutorial and share it here in the comments section when ready.  I have a YouTube channel with technical tutorials regarding bitcoin and the lightning network, again I will share then details once I have the information ready.

Thank you again for all the POW.

Sincerely,

**Comment by Bayernatoor** [comment]
@jairunet This isn't exactly related but I was having the same issues. I ended up finding the change outputs index by running. `bitcoin-cli decoderawtransaction`. I went through the flow on RTL and set the fee to 17sat/vb but for some reason it got broadcasted with a 1sat/vb fee. Did you run into anything similar to this? If not I'll open a new issue.

Now I have the ancestor set to 1 sat and the child set at 1 sat :man_facepalming:

**Reply by jairunet** [reply]
Hello @Bayernatoor,

I did not run into that issue, it looks like something in the process you followed was not accurate.  You can see the screenshot I submitted above in the previous post where I successfully bumped the fees from 1sat/vB to 10sat/vB.

I will be sharing the details here in the comments as soon as I have a video tutorial ready about how to complete the fee bump following the Child-Pays-For-Parent (CPFP) option in RTL.

Thank you for the message and reach out on my YouTube channel or telegram if you are still stuck with bumping the fees.

My contact information > linktr.ee/jairunet

Sincerely,

**Reply by saubyk** [reply]
> @jairunet This isn't exactly related but I was having the same issues. I ended up finding the change outputs index by running. `bitcoin-cli decoderawtransaction`. I went through the flow on RTL and set the fee to 17sat/vb but for some reason it got broadcasted with a 1sat/vb fee. Did you run into anything similar to this? If not I'll open a new issue.
> 
> Now I have the ancestor set to 1 sat and the child set at 1 sat 🤦‍♂️

Hi @Bayernatoor you can bump the new change output again, if your transaction is still stuck. Would recommend doing it via core lightning cli instead. Will investigate if there is an issue in the fee rate conversion from the UI

**Reply by Bayernatoor** [reply]
@saubyk Hey thanks for the response. What's interesting is I did attempt to respend the original RBF using the lightning-cli (I accidentally said bitcoin-cli above) and I got the same message "UTXO .... already reserved". That being said I gave it another try today and it worked with a higher fee rate, I used "urgent" instead of setting a custom rate. 

So I am not exactly sure what happened but if I ever reproduce this I'll make sure to open an issue :) ty!

**Reply by joelmackie** [reply]
Hello, I am having this same issue in finding the Output index.  Currently I have three channels that are stuck in pending open status due to fees being set too low.  Need to bump the fees in order to get these channels open.  Detailed instructions would be greatly appreciated.

**Reply by saubyk** [reply]
hi @joelmackie the heuristic is pretty simple to figure out the change output for your channel opening transaction.
You should read the instructions carefully on the window which opens up for bumping up the transaction.

The requirement to bump up channel open is:

1. Channel opening transaction must have a change output
2. The change output amount must be substantial, so that you can spend it again and the transaction is accepted in the mempool

If you want to figure out what the change output for your transaction is, you need put the channel opening transaction ID in the mempool and look for the output which isn't the channel opening amount.

**Comment by joelmackie** [comment]
Hi Suheb, Reading the instructions I am still having a difficult time understanding exactly how this works. In looking at the transaction on Mempool I see the input&outputs. There are three for this transaction. I see the input and 2 outputs which one is the channel opening amount. How is the index value of the change output found in the block explorer?

**Reply by ShahanaFarooqui** [reply]
![IMG_3716](https://github.com/Ride-The-Lightning/RTL/assets/23240721/a4e0728f-1804-4c96-a68e-2493f05dc41f)

Output index is not mentioned anywhere but it is just an array with starting index as 0. As shown in sample transaction above in yellow

**Reply by joelmackie** [reply]
This is a huge help!  Thank you so much!

**Comment by WantClue** [comment]
is there any detailed instruction on how to achieve a bump up fee for the second time? Running Start9os with cln and rtl opened a channel with way too low fee and did the bump fee feature, but my tx is still stuck and way too low. Apperently I can't bump up using the UI again 🤔

**Reply by saubyk** [reply]
Hello, mempool fee rates are pretty low right now, hope you don't need to fee bump the transaction now

But, to bump it up again, you'd have to do it via the command line. can do it only once via the app.

**Comment by svewa** [comment]
The wording is also very ambiguous. I tried to bump my inital 3sat/vByte fee to 14. Turns out I just enter the fee for the newly create TX. As my opening TX has 6.38kB (3.55kvB), with 37 inputs and 28 outputs.
So I believe either RTL should use the combined size of the opening and CPFP TX, or make it clear to the user that he has to do some calculations here...
